### PR TITLE
Add table azure_cognitive_account. Closes #283

### LINKED ISSUE #283: Add table azure_cognitive_account
**References**
https://docs.microsoft.com/en-us/rest/api/cognitiveservices/accountmanagement/accounts/get
(Required NIST SP 800 53 Rev5)
we need private-endpoint-connections details
https://docs.microsoft.com/en-us/rest/api/cognitiveservices/accountmanagement/private-endpoint-connections/list
Note: This table is required as for NIST SP 800 53 Rev 5 controls.

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro azure-test % ./tint.js azure_cognitive_account
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_cognitive_account []

PRETEST: tests/azure_cognitive_account

TEST: tests/azure_cognitive_account
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_cognitive_account.named_test_resource will be created
  + resource "azurerm_cognitive_account" "named_test_resource" {
      + endpoint             = (known after apply)
      + id                   = (known after apply)
      + kind                 = "Face"
      + location             = "eastus"
      + name                 = "turbottest56407"
      + primary_access_key   = (sensitive value)
      + resource_group_name  = "turbottest56407"
      + secondary_access_key = (sensitive value)
      + sku_name             = "S0"
      + tags                 = {
          + "name" = "turbottest56407"
        }
    }

  # azurerm_resource_group.named_test_resource will be created
  + resource "azurerm_resource_group" "named_test_resource" {
      + id       = (known after apply)
      + location = "eastus"
      + name     = "turbottest56407"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + kind               = "Face"
  + region             = "eastus"
  + resource_aka       = (known after apply)
  + resource_aka_lower = (known after apply)
  + resource_id        = (known after apply)
  + resource_name      = "turbottest56407"
  + sku_name           = "S0"
  + subscription_id    = "d46d7416-f95f-4771-bbb5-529d4c76659c"
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 1s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest56407]
azurerm_cognitive_account.named_test_resource: Creating...
azurerm_cognitive_account.named_test_resource: Still creating... [10s elapsed]
azurerm_cognitive_account.named_test_resource: Creation complete after 12s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest56407/providers/Microsoft.CognitiveServices/accounts/turbottest56407]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

kind = "Face"
region = "eastus"
resource_aka = "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest56407/providers/Microsoft.CognitiveServices/accounts/turbottest56407"
resource_aka_lower = "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest56407/providers/microsoft.cognitiveservices/accounts/turbottest56407"
resource_id = "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest56407/providers/Microsoft.CognitiveServices/accounts/turbottest56407"
resource_name = "turbottest56407"
sku_name = "S0"
subscription_id = "d46d7416-f95f-4771-bbb5-529d4c76659c"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest56407/providers/Microsoft.CognitiveServices/accounts/turbottest56407",
    "kind": "Face",
    "name": "turbottest56407",
    "region": "eastus",
    "resource_group": "turbottest56407",
    "sku": {
      "name": "S0"
    },
    "subscription_id": "d46d7416-f95f-4771-bbb5-529d4c76659c",
    "tags": {
      "name": "turbottest56407"
    },
    "type": "Microsoft.CognitiveServices/accounts"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest56407/providers/Microsoft.CognitiveServices/accounts/turbottest56407",
    "kind": "Face",
    "name": "turbottest56407",
    "region": "eastus",
    "resource_group": "turbottest56407",
    "sku": {
      "name": "S0"
    },
    "subscription_id": "d46d7416-f95f-4771-bbb5-529d4c76659c",
    "tags": {
      "name": "turbottest56407"
    },
    "type": "Microsoft.CognitiveServices/accounts"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest56407/providers/Microsoft.CognitiveServices/accounts/turbottest56407",
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest56407/providers/microsoft.cognitiveservices/accounts/turbottest56407"
    ],
    "name": "turbottest56407",
    "tags": {
      "name": "turbottest56407"
    },
    "title": "turbottest56407"
  }
]
✔ PASSED

POSTTEST: tests/azure_cognitive_account

TEARDOWN: tests/azure_cognitive_account

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  id,
  kind,
  type,
  provisioning_state
from
  azure_cognitive_account;
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------
| name                  | id                                                                                                                                                                | kind          
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------
| test-speech-service-1 | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/Microsoft.CognitiveServices/accounts/test-speech-service-1 | SpeechServices
| test-anomaly-1        | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/Microsoft.CognitiveServices/accounts/test-anomaly-1        | AnomalyDetecto
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------
> select
  name,
  id,
  kind,
  type,
  provisioning_state,
  public_network_access
from
  azure_cognitive_account
where
  public_network_access = 'Enabled';
+----------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+-----------
| name           | id                                                                                                                                                         | kind            | type      
+----------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+-----------
| test-anomaly-1 | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/Microsoft.CognitiveServices/accounts/test-anomaly-1 | AnomalyDetector | Microsoft.
+----------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+-----------
> select
  name,
  id,
  connections ->> 'ID' as connection_id,
  connections ->> 'Name' as connection_name,
  connections ->> 'PrivateEndpointID' as property_private_endpoint_id,
  jsonb_pretty(connections -> 'PrivateLinkServiceConnectionState') as property_private_link_service_connection_state,
  connections ->> 'Type' as connection_type
from
  azure_cognitive_account,
  jsonb_array_elements(private_endpoint_connections) as connections;
+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
| name                | id                                                                                                                                                              | connection_id     
+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
| test-speech-service | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/Microsoft.CognitiveServices/accounts/test-speech-service | /subscriptions/d46
|                     |                                                                                                                                                                 |                   
|                     |                                                                                                                                                                 |                   
|                     |                                                                                                                                                                 |                   
|                     |                                                                                                                                                                 |                   
| test-speech-service | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/Microsoft.CognitiveServices/accounts/test-speech-service | /subscriptions/d46
|                     |                                                                                                                                                                 |                   
|                     |                                                                                                                                                                 |                   
|                     |                                                                                                                                                                 |                   
|                     |                                                                                                                                                                 |                   
+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
> select
  name,
  id,
  settings ->> 'id' as settings_id,
  settings ->> 'name' as settings_name,
  jsonb_pretty(settings -> 'properties' -> 'logs') as settings_properties_logs,
  jsonb_pretty(settings -> 'properties' -> 'metrics') as settings_properties_metrics,
  settings -> 'properties' ->> 'workspaceId' as settings_properties_workspaceId,
  settings ->> 'type' as settings_type
from
  azure_cognitive_account,
  jsonb_array_elements(diagnostic_settings) as settings;
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------
| name                  | id                                                                                                                                                                | settings_id   
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------
| test-speech-service-1 | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/Microsoft.CognitiveServices/accounts/test-speech-service-1 | /subscriptions
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
|                       |                                                                                                                                                                   |               
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------
> 
```
</details>

### CODE REVIEW

File: azure/table_azure_cognitive_account.go [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+				Transform:   transform.FromField("Account.ID").Transform(idToSubscriptionID),
+			},
+		},
+	}
+}
+
+type AccountInfo struct {
+	Account                    cognitiveservices.Account
+	PrivateEndpointConnections []PrivateEndpointConnectionInfo
+}
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:261]: As discussed, update the custom structure

**Comment by ParthaI** [comment]
@c0d3r-arnab can you please check why the lint is failing? https://github.com/turbot/steampipe-plugin-azure/pull/301/checks?check_run_id=3653019352

File: azure/table_azure_cognitive_account.go [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "kind",
+				Description: "The kind of the resource.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "provisioning_state",
+				Description: "Gets the status of the cognitive services account at the time the operation was called. Possible values include: 'Accepted', 'Creating', 'Deleting', 'Moving', 'Failed', 'Succeeded', 'ResolvingDNS'.",
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:47]:
```suggestion
				Description: "The status of the cognitive services account at the time the operation was called. Possible values include: 'Accepted', 'Creating', 'Deleting', 'Moving', 'Failed', 'Succeeded', 'ResolvingDNS'.",
```

File: azure/table_azure_cognitive_account.go [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+			},
+			{
+				Name:        "custom_sub_domain_name",
+				Description: "The subdomain name used for token-based authentication.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Properties.CustomSubDomainName"),
+			},
+			{
+				Name:        "date_created",
+				Description: "Gets the date of cognitive services account creation.",
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:64]:
```suggestion
				Description: "The date of cognitive services account creation.",
```

File: azure/table_azure_cognitive_account.go [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+			{
+				Name:        "etag",
+				Description: "The resource etag.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "is_migrated",
+				Description: "Checks if the resource is migrated from an existing key.",
+				Type:        proto.ColumnType_BOOL,
+				Default:     false,
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:90]: IsMigrated is bool type in API, does not need to mention the default value here.

File: azure/table_azure_cognitive_account.go [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+				Name:        "public_network_access",
+				Description: "Whether or not public endpoint access is allowed for this account. Value is optional but if passed in, must be 'Enabled' or 'Disabled'. Possible values include: 'Enabled', 'Disabled'.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Properties.PublicNetworkAccess"),
+			},
+			{
+				Name:        "restore",
+				Description: "Checks if restore is enabled for the resource.",
+				Type:        proto.ColumnType_BOOL,
+				Default:     false,
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:109]: same as IsMigrated

File: azure/table_azure_cognitive_account.go [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+				Transform:   transform.FromField("Properties.Restore"),
+			},
+			{
+				Name:        "restrict_outbound_network_access",
+				Description: "Checks if outbound network access is restricted for the resource.",
+				Type:        proto.ColumnType_BOOL,
+				Default:     false,
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:116]: same as IsMigrated

File: azure/table_azure_cognitive_account.go [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+			},
+			{
+				Name:        "capabilities",
+				Description: "Gets the capabilities of the cognitive services account. Each item indicates the capability of a specific feature. The values are read-only and for reference only.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Properties.Capabilities"),
+			},
+			{
+				Name:        "diagnostic_settings",
+				Description: "A list of active diagnostic settings for the cognitive service.",
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:145]:
```suggestion
				Description: "A list of active diagnostic settings for the cognitive service account.",
```

File: azure/table_azure_cognitive_account.go [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+			},
+			{
+				Name:        "call_rate_limit",
+				Description: "The call rate limit of the resource.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Properties.CallRateLimit"),
+			},
+			{
+				Name:        "capabilities",
+				Description: "Gets the capabilities of the cognitive services account. Each item indicates the capability of a specific feature. The values are read-only and for reference only.",
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:139]:
```suggestion
				Description: "The capabilities of the cognitive services account. Each item indicates the capability of a specific feature. The values are read-only and for reference only.",
```

File: azure/table_azure_cognitive_account.go [hunk 9]
```diff
@@ -0,0 +1,407 @@
...
+				Transform:   transform.FromField("Properties.QuotaLimit"),
+			},
+			{
+				Name:        "sku",
+				Description: "The resource model definition representing SKU.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "sku_change_info",
+				Description: "Sku change info of account.",
```
Comment (bigdatasourav) [inline azure/table_azure_cognitive_account.go:192]:
```suggestion
				Description: "Sku change info of the resource.",
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**